packages/wdio-cli/src/interface.js

```diff
@@ -55,19 +48,58 @@ export default class WDIOCLInterface extends EventEmitter {
              */
             worker: {}
         }
-        this.clearConsole()
+
+        this.onStart()
+    }
+
+    onStart() {
+        this.log(chalk.bold(`\nExecution of ${chalk.blue(this.totalWorkerCnt)} spec files started at`), this.start.toISOString())
+        if (this.inDebugMode) {
+            this.log(chalk.bgYellow.black('DEBUG mode enabled!'))
+        }
+        if (this.isWatchMode) {
+            this.log(chalk.bgYellow.black('WATCH mode enabled!'))
+        }
+        this.log('')
     }
 
-    clearConsole () {
-        this.display = []
+    onSpecRunning(cid) {
+        this.onJobComplete(cid, this.jobs.get(cid), 0, chalk.bold.cyan('RUNNING'))
+    }
+
+    onSpecRetry(cid, job, retries) {
+        this.onJobComplete(cid, job, retries, chalk.bold.yellow('RETRYING'))
+    }
+
+    onSpecPass(cid, job, retries) {
+        this.onJobComplete(cid, job, retries, chalk.bold.green('PASSED'))
+    }
+
+    onSpecFailure(cid, job, retries) {
+        this.onJobComplete(cid, job, retries, chalk.bold.red('FAILED'))
+    }
+
+    onJobComplete(cid, job, retries, message) {
+        const details = []
+        if (job) {
+            const filename = Array.isArray(job.specs) ? job.specs.join(', ').replace(process.cwd(), '') : ''
```

Comment:
I think it makes sense to create a function for getting `filename` and then remove the inline variable completely. In this case we will get cleaner main function and remove condition in line 87.
```js
function filename({specs}) {
   const result = Array.isArray(specs) ? specs.join(', ').replace(process.cwd(), '') : '';
   return result ? `- ${result}` : ''
}
```
after it line 87 would look like:
`details.push('in', getRunnerName(job.caps), filename(job), retryCount)`

Reply:
@CrispusDH please check again